niworkflows/interfaces/nibabel.py

```diff
@@ -72,20 +79,115 @@ class Binarize(SimpleInterface):
     def _run_interface(self, runtime):
         img = nb.load(self.inputs.in_file)
 
-        self._results['out_file'] = fname_presuffix(
-            self.inputs.in_file, suffix='_masked', newpath=runtime.cwd)
-        self._results['out_mask'] = fname_presuffix(
-            self.inputs.in_file, suffix='_mask', newpath=runtime.cwd)
+        self._results["out_file"] = fname_presuffix(
+            self.inputs.in_file, suffix="_masked", newpath=runtime.cwd
+        )
+        self._results["out_mask"] = fname_presuffix(
+            self.inputs.in_file, suffix="_mask", newpath=runtime.cwd
+        )
 
         data = img.get_fdata()
         mask = data > self.inputs.thresh_low
         data[~mask] = 0.0
         masked = img.__class__(data, img.affine, img.header)
-        masked.to_filename(self._results['out_file'])
+        masked.to_filename(self._results["out_file"])
 
-        img.header.set_data_dtype('uint8')
-        maskimg = img.__class__(mask.astype('uint8'), img.affine,
-                                img.header)
-        maskimg.to_filename(self._results['out_mask'])
+        img.header.set_data_dtype("uint8")
+        maskimg = img.__class__(mask.astype("uint8"), img.affine, img.header)
+        maskimg.to_filename(self._results["out_mask"])
 
         return runtime
+
+
+class _SplitSeriesInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc="input 4d image")
+    allow_3D = traits.Bool(
+        False, usedefault=True, desc="do not fail if a 3D volume is passed in"
+    )
+
+
+class _SplitSeriesOutputSpec(TraitedSpec):
+    out_files = OutputMultiObject(File(exists=True), desc="output list of 3d images")
+
+
+class SplitSeries(SimpleInterface):
+    """Split a 4D dataset along the last dimension into a series of 3D volumes."""
+
+    input_spec = _SplitSeriesInputSpec
+    output_spec = _SplitSeriesOutputSpec
+
+    def _run_interface(self, runtime):
+        filenii = nb.squeeze_image(nb.load(self.inputs.in_file))
+        filenii = filenii.__class__(
+            np.squeeze(filenii.dataobj), filenii.affine, filenii.header
+        )
```

Comment:
The reason that non-terminal 1-dimensions are left in place by `squeeze_image` is to preserve the meaning of each dimension. For example, if I have a time series of a single slice `(64, 64, 1, 48)`, `squeeze_image` will preserve the meaning of `i, j, k, n`, but `np.squeeze` will recast `n` as `k`.

What's the use case that you're taking care of here?

Reply:
Yes, the use case I'm contemplating is separating out the three components of deformation fields and other model-based nonlinear transforms. There is one example of this in the tests.

Reply:
Found it. I guess I would think splitting along the fifth dimension is a different task from splitting along the fourth, but I see that it's convenient to have a single interface. I'm not sure that risking dropping meaningful dimensions is a good idea.

What about forcing to 4D like:

```Python
extra_dims = tuple(dim for dim in img.shape[3:] if dim > 1) or (1,)
if len(extra_dims) != 1:
    raise ValueError("Invalid shape")
img = img.__class__(img.dataobj.reshape(img.shape[:3] + extra_dims),
                    img.affine, img.header)
```

This coerces a 3D image to (x, y, z, 1) and a 4+D image to (x, y, z, n) assuming that dimensions 4-7 are all 1, n or absent.

Reply:
Is it guaranteed that the spatial dimensions will not be affected on that reshape? If so, I'm down with this solution.

Reply:
Trivial dimensions have no effect on indexing, whether it's C- or Fortran ordered. If you don't change the order, it's fine.